Add guide to create slash commands (#76)
LordOfPolls/Dis-Snek: docs/Guides/3 Creating Commands.md

async def my_command_function(ctx: InteractionContext):
    await ctx.send("Hello World")

@slash_command(name="my_long_command", description="My second command :)")
async def my_long_command_function(ctx: InteractionContext):
    # need to defer it, otherwise it fails
    await ctx.defer()
    
    # do stuff for a bit
    await asyncio.sleep(600)
    
    await ctx.send("Hello World")
```

Interactions can either be global, or limited to specific guilds. 
Global commands take up to an hour to sync with Discord and show up, so don't worry when you first register a command.
When testing, it is recommended to use non-global commands, as they sync instantly.

You define non-global commands by passing a list of guild ids to `scopes` in the interaction creation.
```python
@slash_command(name="my_command", description="My first command :)", scopes=[870046872864165888])
async def my_command_function(ctx: InteractionContext):
    await ctx.send("Hello World")
```

For more information, please visit the API reference [here](/API Reference/models/Internal Models/application commands/#dis_snek.models.application_commands.slash_command).

> ## Sub Commands

Reviewer: ```suggestion
## Subcommands
```

> If you have multiple commands that fit under the same category, sub commands are perfect for you.

Reviewer: ```suggestion
If you have multiple commands that fit under the same category, subcommands are perfect for you.
```

Let's define a basic sub command:
```python
@slash_command(
    name="base", 
    description="My command base",
    group_name="group",
    group_description="My command group",
    sub_cmd_name="command",
    sub_cmd_description="My command",
)
async def my_command_function(ctx: InteractionContext):
    await ctx.send("Hello World")
```

> This will show up in discord as `/base group command`. There are two ways to add additional sub commands:

Reviewer: ```suggestion
This will show up in discord as `/base group command`. There are two ways to add additional subcommands:
```

=== ":one: Decorator"
    ```python
    @my_command_function.subcomnmand(sub_cmd_name="second_command", sub_cmd_description="My second command")
    async def my_second_command_function(ctx: InteractionContext):
        await ctx.send("Hello World")
    ```

=== ":two: Repeat Definition"
    ```python
    @slash_command(
        name="base", 
        description="My command base",
        group_name="group",
        group_description="My command group",
        sub_cmd_name="second_command",
        sub_cmd_description="My second command",
    )
    async def my_second_command_function(ctx: InteractionContext):
        await ctx.send("Hello World")
    ```

>     **Note:** This is particullary useful if you want to split sub commands into different files.

Reviewer: ```suggestion
    **Note:** This is particularly useful if you want to split sub commands into different files.
```

If you are using a `OptionTypes.CHANNEL` option, you can restrict the channel a user can choose by setting `channel_types`:
```python
@slash_command(name="my_command", ...)
@slash_option(
    name="channel_option", 
    description="Channel Option", 
    required=True, 
    opt_type=OptionTypes.CHANNEL, 
    channel_types=ChannelTypes.GUILD_TEXT
)
async def my_command_function(ctx: InteractionContext, channel_option: GUILD_TEXT):
    await channel_option.send("This is a text channel in a guild")
    
    await ctx.send("...")
```

You can also set an upper and lower limit for both `OptionTypes.INTEGER` and `OptionTypes.NUMBER` by setting `min_value` and `max_value`:
```python
@slash_command(name="my_command", ...)
@slash_option(
    name="integer_option", 
    description="Integer Option", 
    required=True, 
    opt_type=OptionTypes.INTEGER, 
    min_value=10, 
    max_value=15
)
async def my_command_function(ctx: InteractionContext, integer_option: int):
    await ctx.send(f"You input {integer_option} which is always between 10 and 15")
```

## But I Want A Choice

If your users ~~are dumb~~ constantly misspell specific strings, it might be wise to set up choices. 

> With choices, the user can no longer freely input whatever they want, instead they must choose from a curated list.

Reviewer: ```suggestion
With choices, the user can no longer freely input whatever they want, instead, they must choose from a curated list.
```

To create a choice, simply fill `choices` in `@slash_option()`. An option can have up to 25 choices:
```python
@slash_command(name="my_command", ...)
@slash_option(
    name="integer_option", 
    description="Integer Option", 
    required=True, 
    opt_type=OptionTypes.INTEGER,
    choices=[
        SlashCommandChoice(name="One", value=1),
        SlashCommandChoice(name="Two", value=2)
    ]
)
async def my_command_function(ctx: InteractionContext, integer_option: int):
    await ctx.send(f"You input {integer_option} which is either 1 or 2")
```

For more information, please visit the API reference [here](/API Reference/models/Internal Models/application commands/#dis_snek.models.application_commands.SlashCommandChoice).

## I Need More Than 25 Choices

> Looks like you want autocomplete options. These dynamically show users choices bases on their input. 

Reviewer: ```suggestion
Looks like you want autocomplete options. These dynamically show users choices based on their input. 
```
